examples/report_user_by_email.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env python
+
+""" Script to illustrate how to retrieve a user from the Duo Admin API using the associated email address"""
+
+from __future__ import absolute_import, print_function
+import sys
+import getpass
+from pprint import pprint
+
+import duo_client
+from six.moves import input
+
+argv_iter = iter(sys.argv[1:])
+
+
+def get_next_arg(prompt, secure=False):
+    """Read information from STDIN, using getpass when sensitive information should not be echoed to tty"""
+    try:
+        return next(argv_iter)
+    except StopIteration:
+        if secure is True:
+            return getpass.getpass(prompt)
+        else:
+            return input(prompt)
+
+
+def main():
+    """ Primary script execution code """
+    # Configuration and information about objects to create.
+    admin_api = duo_client.Admin(
+            ikey=get_next_arg('Admin API integration key ("DI..."): '),
+            skey=get_next_arg('integration secret key: ', secure=True),
+            host=get_next_arg('API hostname ("api-....duosecurity.com"): '),
+    )
+
+    # Retrieve user info from API:
+    email_address = get_next_arg('E-mail address of user to retrieve: ')
+    req_params = {"email": email_address}
+    # There is no get_user_by_email in the duo_client_python library, so we use the generic api_call
```

Comment:
Maybe we could add a function in there? I don't know if this a common thing that people might want.

Reply:
Adding a new function to the library specifically for this use case should be rather trivial. I do not have any data on generalized API usage for user searches via email address to offer an opinion on whether a separate function is warranted. Perhaps adding a new optional parameter to the get_users() method to specify the email address could be an alternate possible update to the library.

Reply:
New PR submitted.